pkg/ingress/controller/openstack/octavia.go

```diff
@@ -356,12 +356,26 @@ func (os *OpenStack) EnsureListener(name string, lbID string, secretRefs []strin
 			opts.ProtocolPort = 443
 			opts.Protocol = "TERMINATED_HTTPS"
 		}
+		if len(listenerAllowedCIDRs) > 0 {
+			opts.AllowedCIDRs = listenerAllowedCIDRs
+		}
 		listener, err = listeners.Create(os.Octavia, opts).Extract()
 		if err != nil {
 			return nil, fmt.Errorf("error creating listener: %v", err)
 		}
 
 		log.WithFields(log.Fields{"lbID": lbID, "listenerName": name}).Info("listener created")
+	} else {
+		if len(listenerAllowedCIDRs) > 0 {
```

Comment:
What if the allowed_cidrs is not changed?

Reply:
You mean that this will occur whether the annotation is updated or not ?

Reply:
I guess if the listenerAllowedCIDRs is not changed in this reconcile cycle
no need to execute any command below as it's kept?

Reply:
should be fixed by https://github.com/kubernetes/cloud-provider-openstack/pull/1704/commits/75f28b380004ee3dba697521d87f92bde4f128e0.